pylint/utils.py

```diff
@@ -433,63 +439,129 @@ def print_full_documentation(self):
+                    }
+
+        print("Pylint checkers' options and switches", file=stream)
+        print("-------------------------------------", file=stream)
+        print("", file=stream)
+        print("Pylint checkers can provide three set of features:", file=stream)
+        print("", file=stream)
+        print("* options that control their execution,", file=stream)
+        print("* messages that they can raise,", file=stream)
+        print("* reports that they can generate.", file=stream)
+        print("", file=stream)
+        print("Below is a list of all checkers and their features.", file=stream)
+        print("", file=stream)
+
+        for checker, info in six.iteritems(by_checker):
+            self._print_checker_doc(checker, info, stream=stream)
+
+    def _print_checker_doc(self,  # pylint: disable=no-self-use
+                           checker_name, info, stream=None):
+
+        if not stream:
+            stream = sys.stdout
+
+        doc = info.get('doc', None)
+        module = info.get('module', None)
+        msgs = info.get('msgs', None)
+        options = info.get('options', None)
+        reports = info.get('reports', None)
+
+        title = '%s checker' % (checker_name.replace("_", " ").title())
+
+        if module:
+            # Provide anchor to link against
+            print(".. _%s:\n" % module, file=stream)
+            # Include module name in title
+            title += " (``%s``)" % module
+        print(title, file=stream)
+        print('~' * len(title), file=stream)
+        print("", file=stream)
+        print("Verbatim name of the checker is ``%s``." % checker_name, file=stream)
+        print("", file=stream)
+        if doc:
+            title = 'Documentation'
+            print(title, file=stream)
+            print('^' * len(title), file=stream)
+            print(cleandoc(doc), file=stream)
+            print("", file=stream)
+        if options:
+            title = 'Options'
+            print(title, file=stream)
+            print('^' * len(title), file=stream)
+            _rest_format_section(stream, None, options)
+            print("", file=stream)
+        if msgs:
+            title = 'Messages'
+            print(title, file=stream)
+            print('^' * len(title), file=stream)
+            for msgid, msg in sorted(six.iteritems(msgs),
+                                     key=lambda kv: (_MSG_ORDER.index(kv[0][0]), kv[1])):
+                msg = build_message_def(checker_name, msgid, msg)
+                print(msg.format_help(checkerref=False), file=stream)
+            print("", file=stream)
+        if reports:
+            title = 'Reports'
+            print(title, file=stream)
+            print('^' * len(title), file=stream)
+            for report in reports:
+                print(':%s: %s' % report[:2], file=stream)
+            print("", file=stream)
+        print("", file=stream)
+
+    def print_plugin_documentation(self, stream=None):
```

Comment:
Is it worth having this as a public API? Since it is new, it can also leave in the Sphinx extension itself.